Convert metrics from HandlerAspect to Middleware #2320

lookingformira opened the pull request:
- convert metrics from HandlerAspect to Middleware
- refactor tests following the feature

/claim #2320

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-http?pullRequest=2334) <br/>All committers have signed the CLA.

lookingformira committed:
feature:
- convert metrics from
 HandlerAspect to Middleware
- refactor tests following the feature
